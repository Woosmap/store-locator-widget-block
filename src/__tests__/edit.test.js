import {render, cleanup} from "@testing-library/react";
import StoreLocatorBlockEdit from '../edit';

global.ResizeObserver = jest.fn().mockImplementation(() => ({
	observe: jest.fn(),
	unobserve: jest.fn(),
	disconnect: jest.fn(),
}))
describe("StoreLocatorBlockEdit component", () => {
	afterEach(cleanup);

	it("matches snapshot", () => {
		const attributes = {
			height: 300,
			latitude: 0,
			longitude: 0,
			zoom: 13,
			themeColor: '#000000',
			apiKey: ''
		};

		const {asFragment} = render(
			<StoreLocatorBlockEdit
				{...{
					attributes,
					setAttributes: jest.fn(),
					clientId: "random-id",
					isSelected: false,
				}}
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});

});
