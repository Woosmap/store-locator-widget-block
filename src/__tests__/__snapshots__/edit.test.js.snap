// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocatorBlockEdit component matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Block: undefined"
    class="block-editor-block-list__block wp-block"
    id="block-undefined"
    role="document"
    tabindex="0"
  >
    <div
      class="components-placeholder"
      style="height: 300px;"
    >
      <div
        aria-hidden="true"
        style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; pointer-events: none; opacity: 0; overflow: hidden; z-index: -1;"
      />
      <div
        class="components-placeholder__label"
      >
        <svg
          viewBox="0 0 1024 771.444"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#4FC3F7"
            height="172.249"
            width="166.721"
            x="427.842"
          />
          <rect
            fill="#5586FF"
            height="166.721"
            width="255.601"
            x="338.962"
            y="172.249"
          />
          <rect
            fill="#3D5AFE"
            height="255.609"
            width="166.721"
            x="172.241"
            y="172.249"
          />
          <rect
            fill="#3949AB"
            height="166.721"
            width="338.962"
            y="427.858"
          />
          <rect
            fill="#FFAB00"
            height="166.721"
            width="172.257"
            x="851.743"
            y="176.865"
          />
          <rect
            fill="#FF5252"
            height="255.609"
            width="166.721"
            x="685.022"
            y="176.865"
          />
          <rect
            fill="#F50057"
            height="166.721"
            width="255.601"
            x="596.142"
            y="432.474"
          />
          <rect
            fill="#C51162"
            height="338.961"
            width="166.721"
            x="429.421"
            y="432.483"
          />
        </svg>
        Block for Woosmap Store Locator Widget
      </div>
      <div
        class="components-placeholder__fieldset"
      >
        <svg
          class="components-spinner css-etcl5r-StyledSpinner ea4tfvq2"
          focusable="false"
          height="16"
          role="presentation"
          viewBox="0 0 100 100"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            class="css-mx768t-SpinnerTrack-commonPathProps ea4tfvq1"
            cx="50"
            cy="50"
            r="50"
            vector-effect="non-scaling-stroke"
          />
          <path
            class="css-7c7ebc-SpinnerIndicator-commonPathProps ea4tfvq0"
            d="m 50 0 a 50 50 0 0 1 50 50"
            vector-effect="non-scaling-stroke"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
